### PR TITLE
KT-64377 - Update Dokka build properties

### PR DESCRIPTION
**Pull request by adam-enko**
Updates a couple of integration test properties to use the DokkaBuildProperties class. 

#### Depends on 

* ~#3419~
* #3418

### CODE REVIEW

File: dokka-integration-tests/gradle/build.gradle.kts
Comment (adam-enko) [inline dokka-integration-tests/gradle/build.gradle.kts:28]: I moved this configuration to the integration test convention plugin. The Maven/CLI integration tests might not use it, but to be it made sense to move as much config as possible into convention plugins to keep the project build scripts smaller.

File: build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt
Comment (IgnatBeresnev) [inline build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt:79]: JFYI this particular property (`dokka_integration_test_is_exhaustive)` can actually be dropped altogether. The versions of Kotlin/Gradle/etc that are run under it are not compatible (and haven't been for 2 years), so the tests fail under this flag. And I don't know if it makes sense to invest time into making it work now, so I'd just drop it and return to it when/if it's needed

File: build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt
Comment (IgnatBeresnev) [inline build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt:61]: A nice thing to have, although I don't think it's used much outside of our CI (GitHub/TC) and especially by external people. 

So if the use of this property is corrected in CI (you now have access to change TC configurations), and you can notify Dokka's maintainers in Slack that this property has been changed, I think there's no need to keep the compatibility like that. This idea applies to other internal things too :)

**Review by IgnatBeresnev — APPROVED** [review]

File: build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt
Comment (whyoleg) [inline build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt:81]: BTW: while we are here, if this is used only in integration tests, may be it's better to rename it to `integration_test.useK2` instead of using the same flag which is used to run `dokka` with K2 - now it feels a bit confusing IMO

**Review by whyoleg — APPROVED** [review]

File: build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt
Comment (adam-enko) [reply build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt:79]: Can that be done in a later PR? :) I just want to focus on updating the Gradle config in this chain of PRs, and the `isExhaustive` variable is still used in code.

File: build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt
Comment (adam-enko) [reply build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt:61]: Can do! I was concerned about forgetting to update these properties in whatever miscellaneous places (which could also be on on developers' machines too) after this chain of PRs is completed, but it's not such a big problem.

File: build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt
Comment (adam-enko) [reply build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt:81]: Good idea. Just to be clear the full path will be `org.jetbrains.dokka.integration_test.useK2`

Comment (whyoleg) [reply build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt]: Note: TC config should be updated to use this new property - after merge to master only

**Comment by adam-enko** [comment]
I've rebased this branch to remove #3419

**Pull request merged**
